src/tools/wdump/wdump.cpp

```diff
@@ -0,0 +1,4142 @@
+{
+    char *chunkdata = ReadChunkData(cload);
+    W3dCompressedMotionChannelStruct *channel = (W3dCompressedMotionChannelStruct *)chunkdata;
+
+    if (channel->Flavor < ANIM_FLAVOR_NEW_VALID) {
+        AddString(data, "Flavor", NewFlavorTypes[channel->Flavor], "string");
+    } else {
+        StringClass str;
+        str.Format("W3D_CHUNK_COMPRESSED_ANIMATION_MOTION_CHANNEL Unknown Flavor Type %x", channel->Flavor);
+        data->unknowndata.Add(str);
+        AddString(data, "Flavor", "Unknown", "string");
+    }
+
+    AddInt8(data, "VectorLen", channel->VectorLen);
+
+    if (channel->Flags <= ANIM_CHANNEL_VIS) {
+        AddString(data, "ChannelType", ChannelTypes[channel->Flags], "string");
+    } else {
+        StringClass str;
+        str.Format("W3D_CHUNK_COMPRESSED_ANIMATION_MOTION_CHANNEL Unknown Animation Channel Type %x", channel->Flags);
+        data->unknowndata.Add(str);
+        AddString(data, "ChnanelType", "Unknown", "string");
+    }
+
+    AddInt32(data, "NumTimeCodes", channel->NumTimeCodes);
+    AddInt16(data, "Pivot", channel->Pivot);
+
+    if (channel->Flavor == ANIM_FLAVOR_NEW_TIMECODED) {
+        uint16_t *keyframes = (uint16_t *)(chunkdata + sizeof(W3dCompressedMotionChannelStruct));
+
+        for (int i = 0; i < channel->NumTimeCodes; i++) {
+            StringClass str;
+            str.Format("KeyFrames[%d]", i);
+            AddInt32(data, str, keyframes[i]);
+        }
+
+        int datalen = channel->VectorLen * channel->NumTimeCodes;
+        int pos = sizeof(W3dCompressedMotionChannelStruct);
+        pos += channel->NumTimeCodes * 2;
+
+        if (channel->NumTimeCodes & 1) {
+            pos += 2;
+        }
+
+        uint32_t *values = (uint32_t *)(chunkdata + pos);
+
+        for (int i = 0; i < datalen; i++) {
+            StringClass str;
+            str.Format("Data[%d]", i);
+            AddInt32(data, str, values[i]);
+        }
+    } else {
+        float *scale = (float *)(chunkdata + sizeof(W3dCompressedMotionChannelStruct));
+        AddFloat(data, "Scale", *scale);
+        float *initial = (float *)(chunkdata + sizeof(W3dCompressedMotionChannelStruct) + 4);
+
+        for (int i = 0; i < channel->VectorLen; i++) {
+            StringClass str;
+            str.Format("Initial[%d]", i);
+            AddFloat(data, str, initial[i]);
+        }
+
+        int count = (cload.Cur_Chunk_Length() - sizeof(W3dCompressedMotionChannelStruct) - 4 - 4 * channel->VectorLen) / 4;
+        uint32_t *values = (uint32_t *)(chunkdata + sizeof(W3dCompressedMotionChannelStruct) + 4 + 4 * channel->VectorLen);
+
+        for (int i = 0; i < count; i++) {
+            StringClass str;
+            str.Format("Data[%d]", i);
+            AddInt32(data, str, values[i]);
+        }
+    }
+
+    delete[] chunkdata;
+}
+
+#define CHUNK(id) \
+    { \
+        ChunkDumper c; \
+        c.name = #id; \
+        c.function = dump##id; \
```

Comment:
You can build function name as `Dump_##id` here, then the above functions would look a bit more gracious as
`Dump_W3D_CHUNK_COMPRESSED_ANIMATION_MOTION_CHANNEL`
instead of
`dumpW3D_CHUNK_COMPRESSED_ANIMATION_MOTION_CHANNEL`

Reply:
fixed.